libp2p/utils/semaphore.nim

```diff
@@ -77,13 +78,14 @@ proc release*(s: AsyncSemaphore) =
     trace "Releasing slot", available = s.count,
                             queue = s.queue.len
 
+    s.count.inc
     if s.queue.len > 0:
       var fut = s.queue[0]
       s.queue.delete(0)
       if not fut.finished():
         fut.complete()
+        s.count.dec
```

Comment:
as general hygiene, it's probably better to mutate the state _before_ "releasing" the work that depends on the state update to have happened, even if de-facto, the work that gets released happens later

Reply:
Yeah, this proc is not async, but better be clean